shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -296,6 +299,15 @@ private void removeView(Object[] args) {
             mFakeWindowRootView.removeView(view);
         }
 
+        private void removeViewImmediate(Object[] args) {
+            if (mFakeWindowRootView == null) {
+                Log.w(TAG, "Embedded view called removeViewImmediate while detached from presentation");
+                return;
+            }
+            View view = (View) args[0];
+            mFakeWindowRootView.removeView(view);
```

Comment:
I just looked up the implementation of [`ViewGroup.removeView()`](https://github.com/aosp-mirror/platform_frameworks_base/blob/pie-release/core/java/android/view/ViewGroup.java#L5355).

Iff `view.getAnimation() != null || (mTransitioningViews != null && mTransitioningViews.contains(view))`, the child's `onDetachedFromWindow` will not be called.
Since there's not transition set on `mFakeWindowRootView`, we can just consider call child's `clearAnimation()` to skip this case. That is to say, `onDetachedFromWindow` will always be called immediately.